### DISCUSSION TITLE
Is it possible to find out through PyMechanical/PyAPI whether the Ansys Mechanical license is active?

### DISCUSSION THREAD

**Discussion by BohdanHorban** [ansys/pymechanical#616] [Q&A] [answered]
Hello team, Is it possible to find out through PyMechanical/PyAPI whether the Ansys Mechanical license is active?

**Comment by dipinknair** [comment]
If you using pymechanical embedding.  after you create app instance you can check with
`app.readonly()`
if the license is active it should give you `False`

in addition to this if you print the mechanical app you created using either `remote` or `embedding` you can see what kind of license is active

**Comment by BohdanHorban** [comment]
Hi @dipinknair, thanks. Could do give an example of how to do this in Ansys Mechanical Scripting?

**Comment by dipinknair** [comment]
> Hello team, Is it possible to find out through PyMechanical/PyAPI whether the Ansys Mechanical license is active?

@BohdanHorban 
you can use `Ansys.ACT.Mechanical.MechanicalAPI.Instance.ReadOnlyMode` to check if its in Readonly state or not in Ansys Mechanical Scripting. for any other Mechanical Scripting related questions please see https://discuss.ansys.com/categories/structures

**Comment by dipinknair** [comment, marked as answer]
@BohdanHorban  Alternatively you can run this below script for seeing active license and version info
```
import clr
clr.AddReference("Ansys.Mechanical.Application")
import Ansys
print(Ansys.Mechanical.Application.ProductInfo.ProductInfoAsString)
```

**Comment by BohdanHorban** [comment]
Hi @dipinknair, thanks. It was very helpful
